ios/keyboardObserver/REAKeyboardEventObserver.m

```diff
@@ -51,9 +51,11 @@ - (void)unsubscribeFromKeyboardEvents:(int)listenerId
 - (void)runUpdater
 {
   if (!_displayLink) {
-    _displayLink = [CADisplayLink displayLinkWithTarget:self selector:@selector(updateKeyboardFrame)];
-    _displayLink.preferredFramesPerSecond = 120; // will fallback to 60 fps for devices without Pro Motion display
-    [_displayLink addToRunLoop:[NSRunLoop mainRunLoop] forMode:NSRunLoopCommonModes];
+    RCTExecuteOnMainQueue(^() {
+      _displayLink = [CADisplayLink displayLinkWithTarget:self selector:@selector(updateKeyboardFrame)];
+      _displayLink.preferredFramesPerSecond = 120; // will fallback to 60 fps for devices without Pro Motion display
+      [_displayLink addToRunLoop:[NSRunLoop mainRunLoop] forMode:NSRunLoopCommonModes];
+    })
   }
   _displayLink.paused = NO;
   [self updateKeyboardFrame];
```

Comment:
The potentially async init means that `_displayLink` might not be available when accessed here. Any reason it couldn't be initialized on init?

Reply:
You're right, I've added conditions everywhere

Reply:
Turns out that in some cases we don't need these conditions because when `x` is `nil`, calling `[x someMethod];` does nothing (and, in particular, doesn't crash).